fhir-server-test/src/test/java/com/ibm/fhir/server/test/SearchPerformaceTest.java

```diff
@@ -0,0 +1,488 @@
+                    patient = (Patient) entry.getResource();
+                }
+            }
+        }
+        assertNotNull(observation);
+        assertNotNull(patient);
+        assertEquals(patientId, patient.getId().getValue());
+        assertEquals("Patient/"
+                + patientId, observation.getSubject().getReference().getValue());
+    }
+
+    @SuppressWarnings("rawtypes")
+    @Test(groups = { "server-search-performance" }, dependsOnMethods = {
+    "testCreatePatientAndObservation" })
+    public void testSearchObservationWithSubjectIncluded_filter_elements()
+            throws Exception {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Observation").queryParam("subject", "Patient/"
+                        + patientId).queryParam("_include", "Observation:subject")
+                .queryParam("_elements", "status", "category", "subject")
+                .queryParam("_count", "100")
+                .queryParam("_page", "1")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .header("X-FHIR-TENANT-ID", tanantName)
+                .get();
+        assertResponse(response, Response.Status.OK.getStatusCode());
+        Bundle bundle = response.readEntity(Bundle.class);
+        assertNotNull(bundle);
+        Coding subsettedTag =
+                Coding.builder().system(uri("http://terminology.hl7.org/CodeSystem/v3-ObservationValue")).code(Code.of("SUBSETTED")).display(string("subsetted")).build();
+        assertTrue(FHIRUtil.hasTag(bundle, subsettedTag));
+
+        boolean result = false;
+        for (Bundle.Entry entry : bundle.getEntry()) {
+
+            if (DEBUG_SEARCH) {
+
+                SearchAllTest.generateOutput(entry.getResource());
+                System.out.println(result + " "
+                        + FHIRUtil.hasTag(entry.getResource(), subsettedTag));
+            }
+
+            result = result
+                    || FHIRUtil.hasTag(entry.getResource(), subsettedTag);
+        }
+        assertTrue(result);
+
+        assertTrue(bundle.getEntry().size() == 2);
+
+        Observation observation = null;
+        Patient patient = null;
+        for (Bundle.Entry entry : bundle.getEntry()) {
+            if (entry.getResource() != null) {
+                if (entry.getResource() instanceof Observation) {
+                    observation = (Observation) entry.getResource();
+                } else if (entry.getResource() instanceof Patient) {
+                    patient = (Patient) entry.getResource();
+                }
+            }
+        }
+        assertNotNull(observation);
+        assertTrue(FHIRUtil.hasTag(observation, subsettedTag));
+        // Verify that only the requested elements (and "mandatory elements") are
+        // present in the returned Observation.
+        Method[] observationMethods = Observation.class.getDeclaredMethods();
```

Comment:
Is Reflection required here? ... I get why it's here, but it feels a little dirty. 
FHIRPath may be another option which would simplify. 

Also I get this is testCode.

Reply:
those codes were copied from SearchTTest and modified based on that

Reply:
OK - makes sense... I get it's test Code.